src/java/org/apache/cassandra/db/memtable/PersistentMemoryMemtable.java

```diff
@@ -405,6 +405,31 @@ public void performSnapshot(String snapshotName)
     {
         // TODO: implement. Figure out how to restore snapshot (with external tools).
     }
+    @Override
+    public void performGarbageCollect()
+    {
+        logger.info("Starting {} from memtable for {}.{}.", OperationType.GARBAGE_COLLECT, metadata().keyspace, metadata().name);
+        ConcurrentLongART memtableCart = tablesMetadataMap.get(metadata().id).getMemtableCart();
+        long removedTombstones = 0;
+        try (AutoCloseableIterator<LongART.Entry> cartIterator = memtableCart.getEntryIterator())
```

Comment:
Does this make the table unwriteable and unreadable until the operation completes?

If so, this behaviour should be documented and I would consider logging a warning to that effect and think how it could be made interruptible.

Reply:
The entire table is not locked, only the shard containing the partition is locked, writes & reads to other shards can continue. We will add documentation explaining this.